build.rs

```diff
@@ -76,7 +76,7 @@ fn build(metadata: &Metadata) {
 
         // If we're cross-compiling, let configure know.
         if metadata.host != metadata.target {
-            configure_args.push(format!("--host={}", metadata.target));
+            configure_args.push(format!("--host={}", metadata.host));
```

Comment:
I'm afraid that this does not look correct to me. GNU autotools uses "host" and "target" differently than Cargo: https://github.com/lu-zero/autotools-rs/blob/27d0437e071ddbf68ff0bbacfa02e01a04cc6b42/src/lib.rs#L71-L94

Reply:
Thanks for the link, I wasn't aware of this difference before. 

Also I digged bit more on that, noticed that the zigbuild actually set `CC_<triplets>` and `AR_<triplets>` environment variables in, that are not passed all the way down to configure and make, so it was actually using the host tool & target to compile it into a macos target. Updated the build.rs to pass these when set.